common/workunit/workunit.cpp

```diff
@@ -5231,6 +5288,8 @@ void CLocalWorkUnit::copyWorkUnit(IConstWorkUnit *cached, bool all)
         CLocalWUResult result(LINK(&results->query()));
         result.setResultStatus(ResultStatusUndefined);
     }
+
+    copyTree(p, fromP, "usedsources"); // field usage
```

Comment:
@richardkchapman   ESP services will use this information to verify a user has access to all the fields before we submit the WU. I don't want to put it in the helper SO because then we have to wait until runtime, and the engines don's access LDAP anyway. I think we should make sure Cassandra supports this feature